src/replication.c

```diff
@@ -394,7 +397,9 @@ remove_replica_from_list(spec_t *spec, int iofd) {
 			pthread_cond_signal(&replica->r_cond);
 			MTX_UNLOCK(&replica->r_mtx);
 			spec->replica_count--;
-			if(spec->replica_count == 0){
+			if((spec->healthy_rcount < spec->consistency_factor) &&
```

Comment:
condition to allow IOs should be: (n is replication factor, m is consistency factor)
- MAX(n-m+1, m) number of degraded replicas (or)
- 'm' number of connected replicas with atleast one healthy replica

Reply:
first condition is edited